modules/cart/src/Plugin/QueueWorker/CartExpire.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace Drupal\commerce_cart\Plugin\QueueWorker;
+
+use Drupal\commerce_order\Entity\OrderInterface;
+use Drupal\Core\Queue\QueueWorkerBase;
+
+/**
+ * Removes an expired Cart.
+ *
+ * @QueueWorker(
+ *  id = "cart_expiries",
+ *  title = @Translation("Commerce Cart expiration"),
+ *  cron = {}
+ * )
+ */
+class CartExpire extends QueueWorkerBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function processItem($data) {
+    $orders = \Drupal::entityTypeManager()->getStorage('commerce_order')
+      ->loadMultiple($data);
+    foreach ($orders as $order) {
+      // Ensure that orders slated for clearing have not been completed since
+      // they were last queued.
+      if ($order instanceof OrderInterface && is_null($order->getCompletedTime())) {
+        $order_type_storage = \Drupal::entityTypeManager()
+          ->getStorage('commerce_order_type');
+        $order_type = $order_type_storage->load($order->bundle());
+        $elapsed = REQUEST_TIME - $order->getCreatedTime();
```

Comment:
We actually have a `time` service! Because REQUEST_TIME can't be trusted in long running ops (CLI). It's provided by core in 8.3.x, but we made our own. You can find it in the commerce.services.yml